versioned/spi/src/main/java/org/projectnessie/versioned/Key.java

```diff
@@ -58,11 +46,11 @@ public final int compareTo(Key that) {
 
   @Override
   public int hashCode() {
-    final Collator collator = COLLATOR.get();
-    return getElements().stream()
-        .map(collator::getCollationKey)
-        .collect(Collectors.toList())
-        .hashCode();
+    int h = 1;
+    for (String element : getElements()) {
+      h = 31 + h + element.hashCode();
```

Comment:
Does this match the old hash code or just an unrelated hash function? If the latter, wouldn't you normally want to _multiply_ by 31 here?